internal/app/wwctl/node/edit/main.go

```diff
@@ -20,120 +15,140 @@ import (
 )
 
 func CobraRunE(cmd *cobra.Command, args []string) error {
-	canWrite, err := apiutil.CanWriteConfig()
-	if err != nil {
-		return fmt.Errorf("while checking whether can write config, err: %w", err)
-	}
-	if !canWrite.CanWriteConfig {
-		return fmt.Errorf("can not write to config exiting")
+	if !node.CanWriteConfig() {
+		return fmt.Errorf("can not write to config: exiting")
 	}
+
 	editor := os.Getenv("EDITOR")
 	if editor == "" {
 		editor = "/bin/vi"
 	}
-	if len(args) == 0 {
-		args = append(args, ".*")
+	wwlog.Debug("using editor: %s", editor)
+
+	registry, regErr := node.New()
+	if regErr != nil {
+		return regErr
 	}
-	filterList := wwapiv1.NodeList{
-		Output: args,
+
+	if len(args) == 0 {
+		for nodeID := range registry.Nodes {
+			args = append(args, nodeID)
+		}
 	}
-	nodeListMsg := apinode.FilteredNodes(&filterList)
-	nodeMap := make(map[string]*node.Node)
-	// got proper yaml back
-	_ = yaml.Unmarshal([]byte(nodeListMsg.NodeConfMapYaml), nodeMap)
-	file, err := os.CreateTemp(os.TempDir(), "ww4NodeEdit*.yaml")
-	if err != nil {
-		return fmt.Errorf("could not create temp file: %s", err)
+	wwlog.Debug("node list: %v", args)
+
+	tempFile, tempErr := os.CreateTemp(os.TempDir(), "ww4NodeEdit*.yaml")
+	if tempErr != nil {
+		return fmt.Errorf("could not create temp file: %s", tempErr)
 	}
-	defer os.Remove(file.Name())
-	for {
-		_ = file.Truncate(0)
-		_, _ = file.Seek(0, 0)
-		if !NoHeader {
-			yamlTemplate := node.UnmarshalConf(node.Node{}, []string{"tagsdel"})
-			_, _ = file.WriteString("#nodename:\n#  " + strings.Join(yamlTemplate, "\n#  ") + "\n")
+	defer os.Remove(tempFile.Name())
+
+	if !NoHeader {
+		yamlTemplate := node.UnmarshalConf(node.Node{}, nil)
+		if _, err := tempFile.WriteString("#nodename:\n#  " + strings.Join(yamlTemplate, "\n#  ") + "\n"); err != nil {
+			return err
 		}
-		_, _ = file.WriteString(nodeListMsg.NodeConfMapYaml)
-		_, _ = file.Seek(0, 0)
-		hasher := sha256.New()
-		if _, err := io.Copy(hasher, file); err != nil {
-			wwlog.Error("Problems getting checksum of file %s\n", err)
+	}
+
+	origNodes := make(map[string]*node.Node)
+	for _, nodeID := range args {
+		if n, ok := registry.Nodes[nodeID]; ok {
+			origNodes[nodeID] = n
 		}
-		sum1 := hex.EncodeToString(hasher.Sum(nil))
-		err = util.ExecInteractive(editor, file.Name())
-		if err != nil {
+	}
+
+	if origYaml, err := util.EncodeYaml(origNodes); err != nil {
+		return err
+	} else if _, err := tempFile.Write(origYaml); err != nil {
+		return err
+	}
+
+	sum1, sumErr := util.HashFile(tempFile)
+	if sumErr != nil {
+		return sumErr
+	}
+	wwlog.Debug("original hash: %s", sum1)
+
+	for {
+		if err := util.ExecInteractive(editor, tempFile.Name()); err != nil {
 			return fmt.Errorf("editor process existed with non-zero")
 		}
-		_, _ = file.Seek(0, 0)
-		hasher.Reset()
-		if _, err := io.Copy(hasher, file); err != nil {
-			wwlog.Error("Problems getting checksum of file %s\n", err)
+
```

Comment:
Can you please output the value of `err` here? So if `/bin/vi` isn't on the system I just get a message that editor process failed and no reason why this happens,